### PR TITLE
Change createOrOverwrite to accept file content

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsOutputFile.java [hunk 1]
```diff
@@ -69,7 +68,7 @@ public OutputStream createOrOverwrite(AggregatedMemoryContext memoryContext)
     }
 
     @Override
-    public void createExclusive(Slice content)
+    public void createExclusive(byte[] data)
```
Comment (dain) [inline lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsOutputFile.java:64]: Consider adding a version with data, offset, and length, and a default for just data?  We can do this later also.

File: lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoOutputFile.java [hunk 2]
```diff
@@ -25,25 +25,26 @@
...
     }
 
-    default OutputStream createOrOverwrite()
-            throws IOException
-    {
-        return createOrOverwrite(newSimpleAggregatedMemoryContext());
-    }
+    /**
+     * Create file with the specified content, atomically if possible.
+     * The file will be replaced if it already exists.
```
Comment (dain) [inline lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoOutputFile.java:38]: Describe the state after a failure.  I believe the answer is anything is possible, exist, not exist, truncate existing, maybe half written?

File: lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoOutputFile.java [hunk 3]
```diff
@@ -25,25 +25,26 @@
...
+     */
+    void createOrOverwrite(byte[] data)
+            throws IOException;
 
     /**
-     * Create file exclusively and atomically with specified contents.
+     * Create file exclusively and atomically with the specified content.
```
Comment (dain) [inline lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoOutputFile.java:44]: describe possible states after failure here also

**Review by dain — APPROVED** [review]

File: lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsOutputFile.java (hunk 1 above)
Comment (electrum) [reply lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsOutputFile.java:64]: None of the callers need this. We can add later if needed.

**Pull request merged**